src/OrchardCore/OrchardCore.DisplayManagement.Liquid/LiquidViewTemplate.cs

```diff
@@ -107,16 +107,21 @@ internal class ShapeAccessor : DelegateAccessor
 
         private static Func<object, string, object> _getter => (o, n) =>
         {
-            if ((o as Shape).Properties.TryGetValue(n, out object result))
+            if (o is Shape shape)
             {
-                return result;
-            }
+                if (shape.Properties.TryGetValue(n, out object result))
+                {
+                    return result;
+                }
 
-            foreach (var item in (o as Shape).Items)
-            {
-                if (item is IShape && item.Metadata.Type == n)
+                foreach (var item in shape.Items)
                 {
-                    return item;
+                    // The prefix is the Name of the part. It also works for named parts,
+                    // so we can access Model.Content.MyNamedPart
+                    if (item is IShape itemShape && itemShape.Metadata.Prefix == n)
```

Comment:
With some shapes, e.g when `.Type = "Contents__Metadata"`, we have `.Prefix = ""`. With some shapes, we have `.Prefix = null`. So, maybe we should also check `.Type`.

I also see the following in some places where the dot character could be a problem to access properties.

    Prefix = htmlFieldPrefix + "." + Prefix;
    Prefix = context.HtmlFieldPrefix + "." + Prefix;
    Prefix = typePartDefinition.Name + "." + partFieldDefinition.Name;

Humm, maybe we would have the same problem with fields as with named part, but here the prefix contains a dot.

Reply:
Then I will add a `Name` property, which will default to the type, then the name, and can be anything else too.